test/utilities.js

```diff
@@ -232,6 +232,42 @@ describe('utilities', function () {
     expect(expect('foo').result()).to.equal('result');
   });
 
+  it('addMethod returns new assertion with flags copied over', function () {
+    var assertionConstructor;
+
+    chai.use(function(_chai, utils) {
+      assertionConstructor = _chai.Assertion;
+      _chai.Assertion.addMethod('returnNewAssertion', function () {
+        utils.flag(this, 'mySpecificFlag', 'value1');
+        utils.flag(this, 'ultraSpecificFlag', 'value2');
+      });
+    });
+
+    assertion1 = expect('foo');
+    assertion2 = assertion1.to.returnNewAssertion();
+
+    // Checking if a new assertion was returned
+    expect(assertion1).to.not.be.equal(assertion2);
+
+    // Checking if it's really an instance of an Assertion
+    expect(assertion2).to.be.instanceOf(assertionConstructor);
+
+    // If it is, calling length on it should return an assertion, not a function
+    var anAssertion = expect([1, 2, 3]).to.be.an.instanceof(Array);
+    expect(anAssertion.length).to.be.an.instanceof(assertionConstructor);
```

Comment:
Well thought @keithamus.
This fix appears to work with the expression you posted, but I went to the issue to test other examples and this appears to fail:

```js
expect(my_object).to.have.a.deep.property("foo.bar").that.is.an("array").with.a.length.of.at.least(1);
```

The reason is the same you explained on the issue, we are getting back an `assert` function instead of an `Assertion` object.

I'll take a further look at this to find the exact reason.

Thanks again for your feedback, it was really useful :smiley: